datastore/src/main/java/org/jboss/pnc/datastore/DefaultDatastore.java

```diff
@@ -141,6 +142,12 @@ public BuildRecord storeCompletedBuild(
             List<Artifact> dependencies) {
         BuildRecord buildRecord = buildRecordBuilder.build(true);
         logger.debug("Storing completed build {}.", buildRecord);
+        BuildRecord previouslySavedBuild = buildRecordRepository.queryById(buildRecord.getId());
+        if (previouslySavedBuild != null) {
+            throw new IllegalStateException(
+                    "When trying to save build " + buildRecord + " previously saved build " + previouslySavedBuild
```

Comment:
Maybe I would consider adding the dto in the exception message only once if they are the same.

Reply:
they may have different status